tests/flow/test_list.py

```diff
@@ -143,3 +143,171 @@ def test03_null_list_function_inputs(self):
         expected_result = [[False]]
         self.env.assertEquals(actual_result.result_set, expected_result)
 
+    def test04_toBooleanList(self):
+        # NULL input should return NULL
+        expected_result = [None]
```

Comment:
please test
```
MATCH (n) RETURN toBooleanList(n)
MATCH (n) RETURN toBooleanList([n])
MATCH ()-[e]->() RETURN toBooleanList(e)
MATCH ()-[e]->() RETURN toBooleanList([e])
MATCH p=()-[]->() RETURN toBooleanList(p)
MATCH p=()-[]->() RETURN toBooleanList([p])
```

Reply:
Hi, when at least one node is defined, I think that the result is right. 
But what is the expected result without any node defined?

```
127.0.0.1:6379> graph.query g "CREATE (:A {name: 'a'})"
1) 1) "Labels added: 1"
   2) "Nodes created: 1"
   3) "Properties set: 1"
   4) "Cached execution: 0"
   5) "Query internal execution time: 0.594114 milliseconds"
127.0.0.1:6379> graph.query g "MATCH (n) RETURN toBooleanList(n)"
(error) Type mismatch: expected List but was Node
127.0.0.1:6379> graph.query g "MATCH (n) RETURN toBooleanList([n])"
1) 1) "toBooleanList([n])"
2) 1) 1) "[NULL]"
3) 1) "Cached execution: 1"
   2) "Query internal execution time: 0.173370 milliseconds"
```

```
127.0.0.1:6379> graph.query g "MATCH (n) DELETE(n)"
1) 1) "Nodes deleted: 1"
   2) "Cached execution: 0"
   3) "Query internal execution time: 0.357105 milliseconds"
127.0.0.1:6379> graph.query g "MATCH (n) RETURN toBooleanList(n)"
1) 1) "toBooleanList(n)"
2) (empty array)
3) 1) "Cached execution: 1"
   2) "Query internal execution time: 0.127810 milliseconds"
127.0.0.1:6379> graph.query g "MATCH (n) RETURN toBooleanList([n])"
1) 1) "toBooleanList([n])"
2) (empty array)
3) 1) "Cached execution: 1"
   2) "Query internal execution time: 0.159592 milliseconds"
```

Reply:
Neo4j, no nodes:
```
MATCH (n) RETURN toBooleanList(n)
Type mismatch: expected List<T> but was Node (line 1, column 32 (offset: 31))
"MATCH (n) RETURN toBooleanList(n)"
```

```
MATCH (n) RETURN toBooleanList([n])
(no changes, no records)
```

Reply:
Hi, to get a similar result to Neo4j, I think that the changes will have a high impact. 

Because in case of no node, or no matches, the result "(empty array) " should be replaced by a Node (a null Node).

This is the current behavior:
```
127.0.0.1:6379> GRAPH.QUERY g "MATCH (n) RETURN n"
1) 1) "n"
2) (empty array)
```
And a similar change should be done for paths an edges.
```
127.0.0.1:6379> GRAPH.QUERY g "MATCH p=()-[]->() RETURN p"
1) 1) "p"
2) (empty array)
```
```
127.0.0.1:6379> GRAPH.QUERY g "MATCH ()-[e]->() RETURN e"
1) 1) "e"
2) (empty array)
```

Reply:
I believe Neo's response: `(no changes, no records)` is similar to our `(empty array)` this means (in Redis-cli) that the result-set is empty no to confuse with a result-set containing an empty array.